WooCommerce/Classes/Extensions/UINavigationController+Woo.swift

```diff
@@ -57,30 +57,20 @@ extension UIViewController: UINavigationBarBackButtonHandler {
 }
 
 extension UINavigationController: UINavigationBarDelegate {
-
-    // This delegate method is not called on the simulator or device running iOS 13.4 from Xcode.
-    // You need to use a release build.
-    public func navigationBar(_ navigationBar: UINavigationBar, shouldPop item: UINavigationItem) -> Bool {
-        guard let items = navigationBar.items else {
-            return false
-        }
-
-        if viewControllers.count < items.count {
-            return true
-        }
```

Comment:
I removed this check because the check did not make sense to me since there could be many items and this way we would always return `true` and would not call `shouldPopOnBackButton()`

Reply:
Agreed, I tried to make sense of this check and don't understand why it would be needed here. I also found examples of where it leads to the wrong result, so removing it makes sense to me.

That said, since the only thing we're now checking is essentially `topViewController.shouldPopOnBackButton()` should we add a check to make sure the navigation item we're popping corresponds to the top view controller? Something like:

```
guard let vc = topViewController, vc.navigationItem == item else {
    return true
}
```